src/mahoji/commands/tools.ts

```diff
@@ -26,6 +26,16 @@ import {
 } from '@/mahoji/lib/abstracted_commands/stashUnitsCommand.js';
 
 const skillsVals = Object.values(Skills);
+const skillMap = new Map(skillsVals.map(skill => [skill.id, skill] as const));
+const discordIDRegex = /^\d{17,20}$/;
+
+function sanitizeDiscordID(raw: string): string | null {
+	const cleaned = raw.replace(/[^0-9]/g, '');
+	if (!discordIDRegex.test(cleaned)) {
+		return null;
+	}
+	return cleaned;
+}
```

Comment:
We have a `isValidDiscordSnowflake` that would be better to use here, removing the discordIdRegex here.

```ts
import { isValidDiscordSnowflake } from '@oldschoolgg/util';

if (!isValidDiscordSnowflake(inputSnowflake)) return 'Invalid';
```

However, I think it would be better to instead just use a `type: 'User',` instead of a `type: 'String',`, then you just get a user object directly given to you, and for users they dont need to figure out how to get a user id.

Side note: this lets you check anyones xp gains/drystreaks? seems okay to me although someone might complain